### PR TITLE
feat: Description `item` support block

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

https://github.com/ant-design/ant-design/discussions/51364

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Descriptions items support block       |
| 🇨🇳 Chinese |   描述组件支持单行铺满        |

### CODE REVIEW

**Comment by crazyair** [comment]
text 有 diff，是因为这个功能去掉了，本身也是 demo 写的不对，column 是 2，但是 row = 3

image

File: components/descriptions/demo/row.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  {
+    key: '3',
+    label: 'Live',
+    children: 'Hangzhou, Zhejiang',
+  },
+  {
+    key: '4',
+    label: 'Remark',
+    span: 'row',
```
Comment (zombieJ) [inline components/descriptions/demo/row.tsx:24]: 参考其他的 API 命名风格，`block` 应该更适合一些

File: components/descriptions/__tests__/index.test.tsx [hunk 2]
```diff
@@ -78,6 +78,30 @@ describe('Descriptions', () => {
...
+  it('span = row', () => {
+    const { container } = render(
+      <Descriptions
+        column={3}
+        items={[
+          { label: 'Product', children: 'Cloud Database', span: 2 },
+          { label: 'Billing', children: 'Prepaid', span: 1 },
+          { label: 'Time', children: '18:00:00' },
+          { label: 'Product', children: 'Cloud Database', span: 'block' },
+          { label: 'Billing', children: 'Prepaid', span: 'block' },
```
Comment (zombieJ) [inline components/descriptions/__tests__/index.test.tsx:90]: 中间在加一个 span: 1 的看看在有 span 的时候会不会拉长

Comment (crazyair) [reply components/descriptions/__tests__/index.test.tsx]: 加了

Comment (crazyair) [reply components/descriptions/__tests__/index.test.tsx]: 还有最后，加了 span 1，会不会被强制改成 column，这个就不加了

File: components/descriptions/__tests__/index.test.tsx [hunk 3]
```diff
@@ -78,6 +78,32 @@ describe('Descriptions', () => {
...
+          { label: '7', children: '' },
+        ]}
+      />,
+    );
+    expect(container.querySelectorAll('.ant-descriptions-item')[0]).toHaveAttribute('colSpan', '2');
+    expect(container.querySelectorAll('.ant-descriptions-item')[1]).toHaveAttribute('colSpan', '1');
+    expect(container.querySelectorAll('.ant-descriptions-item')[2]).toHaveAttribute('colSpan', '3');
+    expect(container.querySelectorAll('.ant-descriptions-item')[3]).toHaveAttribute('colSpan', '3');
+    expect(container.querySelectorAll('.ant-descriptions-item')[4]).toHaveAttribute('colSpan', '1');
+    expect(container.querySelectorAll('.ant-descriptions-item')[5]).toHaveAttribute('colSpan', '3');
```
Comment (zombieJ) [inline components/descriptions/__tests__/index.test.tsx:102]: 这个不应该是 2 么？

File: components/descriptions/__tests__/index.test.tsx [hunk 4]
```diff
@@ -78,6 +78,32 @@ describe('Descriptions', () => {
...
+          { label: '1', children: '', span: 1 },
+          { label: '2', children: '' },
+          { label: '3', children: '', span: 'filled' },
+          { label: '4', children: '', span: 1 },
+          { label: '5', children: '', span: 'filled' },
+          { label: '6', children: '' },
+          { label: '7', children: '' },
+        ]}
+      />,
+    );
```
Comment (zombieJ) [inline components/descriptions/__tests__/index.test.tsx:96]: | label | span | real span | others |
| - | - | - | - |
| 0 | 2 | 2 | |
| 1 | 1 | 1 | |
| 2 |  | 1 | 默认是 1 |
| 3 | filled | 2 | 2 为 1，3 则填充为 2 |
| 4 | 1 | 1 | |
| 5 | filled | 2 | 4 为 1，5 则填充为 2 |
| 6 |  | 1 | 默认是 1 |
| 7 |  | 2 | 由于为最后一个，被当做 `filled` 填充为 2 |

File: components/descriptions/demo/block.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    children: 'Hangzhou, Zhejiang',
+  },
+  {
+    label: 'Remark',
+    span: 'filled', // span = 3
+    children: 'empty',
+  },
+  {
+    label: 'Address',
+    span: 1, // span = 3
```
Comment (zombieJ) [inline components/descriptions/demo/block.tsx:22]: > // `span` will be `3` and warning for `span` is not align to the end

File: components/descriptions/hooks/useRow.ts [hunk 6]
```diff
@@ -3,65 +3,51 @@ import { useMemo } from 'react';
...
-
-      // Additional handle last one
-      if (index === rowItems.length - 1) {
-        const [item, itemExceed] = getFilledItem(rowItem, rowRestCol, span);
-        exceed = exceed || itemExceed;
-
-        tmpRow.push(item);
+    .forEach((rowItem) => {
+      const { filled, ...restItem } = rowItem;
+      count += rowItem.span || 1;
```
Comment (zombieJ) [inline components/descriptions/hooks/useRow.ts:20]: 这个是不是不应该提前加？看起来应该预先拿到剩余空间和 span 进行计算然后再加。这样逻辑会比较顺一些。

Comment (crazyair) [reply components/descriptions/hooks/useRow.ts:20]: count 算是一个 sum 计数的，count - column 就是剩余的

Comment (zombieJ) [reply components/descriptions/hooks/useRow.ts:20]: 我意思是先 if filled or last one，就直接 count 加上去。然后再按照 span 的来算。

Comment (crazyair) [reply components/descriptions/hooks/useRow.ts:20]: 设置 filled 了，span 是 0 ，先加会先加，都一样吧，要区分也行

**Review by zombieJ — APPROVED** [review]

**Pull request merged**